vtysh/vtysh_config.c

```diff
@@ -155,6 +155,47 @@ static void config_add_line_uniq(struct list *config, const char *line)
 	listnode_add_sort(config, XSTRDUP(MTYPE_VTYSH_CONFIG_LINE, line));
 }
 
+/* Global variable to keep track of the KEYCHAIN_KEY_NODE for comparing */
+static char *key_ptr;
```

Comment:
uh... okay... this raises alarm bells for me but I didn't review it yet.  Maybe it's OK, did someone else double check it?

Reply:
Please try to review the full logic. It's the best way to solve the running config duplication. If you feel there is better logic we can discuss on it.